src/core/credentials/transport/tls/grpc_tls_certificate_provider.cc

```diff
@@ -488,6 +489,79 @@ int64_t FileWatcherCertificateProvider::TestOnlyGetRefreshIntervalSecond()
   return refresh_interval_sec_;
 }
 
+std::shared_ptr<InMemoryCertificateProvider>
+InMemoryCertificateProvider::CreateCertificateProvider(
+    std::string root_certificates, PemKeyCertPairList pem_key_cert_pairs) {
+  return std::make_shared<InMemoryCertificateProvider>(root_certificates,
+                                                       pem_key_cert_pairs);
+}
+
+InMemoryCertificateProvider::InMemoryCertificateProvider(
+    std::string root_certificates, PemKeyCertPairList pem_key_cert_pairs)
+    : distributor_(MakeRefCounted<grpc_tls_certificate_distributor>()),
+      pem_key_cert_pairs_(std::move(pem_key_cert_pairs)),
+      root_certificates_(root_certificates) {
+  distributor_->SetWatchStatusCallback([this](std::string cert_name,
```

Comment:
The logic here should follow the pattern from the file watcher provider, not the logic from the static provider.  that way, it will be easier to eventually allow setting a SPIFFE bundle map for the root cert info.